src/main/java/lotto/domain/NumberGroup.java

```diff
@@ -0,0 +1,63 @@
+package lotto.domain;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class NumberGroup {
+    public static final int LIMIT = 6;
+    private final List<Number> numbers;
+
+    public NumberGroup(List<Number> numbers) {
+        this.numbers = numbers;
+    }
+
+    public boolean appropriate(int count) {
+        return numbers.size() == count;
+    }
+
+    public void match(NumberGroup winningNumbers) {
+        Iterator<Number> thisNumberIterator = numbers.iterator();
+        Iterator<Number> winningNumberIterator = winningNumbers.numbers.iterator();
+
+        int matchResult = 0;
+        while (thisNumberIterator.hasNext() && winningNumberIterator.hasNext()) {
+            matchResult = matchResult(thisNumberIterator.next(), winningNumberIterator.next(), matchResult);
+        }
+        matchMatchCriteria(matchResult);
+    }
+
+    public int matchResult(Number thisNumber, Number winningNumber, int matchResult) {
+        if (thisNumber.equals(winningNumber)) {
+            matchResult++;
+        }
+        return matchResult;
+    }
+
+    public void matchMatchCriteria(int result) {
+        Arrays.stream(MatchCriteria.values())
+                .filter(m -> m.equalsCriteria(result))
+                .findFirst()
+                .map(MatchCriteria::match)
+                .orElse(MatchCriteria.NOTING);
+    }
```

Comment:
리턴 타입이 없다면 ifPresent를 활용해 보시면 좋을 것 같아요.

```suggestion
    public void matchMatchCriteria(int result) {
        Arrays.stream(MatchCriteria.values())
                .filter(m -> m.equalsCriteria(result))
                .findFirst()
                .ifPresent(MatchCriteria::match);
    }
```